tests/run.py

```diff
@@ -2,41 +2,136 @@
 
 import sys
 import argparse
+import random
 
 from helper import Test, TestGroup, run, find_profiles
 from talent_options import talent_combinations
 
 FIGHT_STYLES = ('Patchwerk', 'DungeonSlice', 'HeavyMovement')
 COVENANTS = ('Kyrian', 'Venthyr', 'Night_Fae', 'Necrolord')
 
-parser = argparse.ArgumentParser(description='Run simc tests')
-parser.add_argument('specialization', metavar='spec', type=str,
-                    help='Simc specialization in the form of CLASS_SPEC, eg. Priest_Shadow')
-parser.add_argument('--test-trinkets', default=False, action='store_true',
-                    help='Test trinkets. Trinket tests require simc-support library to be installed.')
-parser.add_argument('--trinkets-fight-style', default='DungeonSlice', type=str,
-                    help='Fight style used for trinket simulations.')
-args = parser.parse_args()
 
-def test_trinkets(klass : str, path : str):
-    if not args.test_trinkets:
+def test_talents(klass: str, path: str):
+    for fight_style in FIGHT_STYLES:
+        grp = TestGroup('{}/{}/talents'.format(profile, fight_style),
+                        fight_style=fight_style, profile=path)
+        tests.append(grp)
+        for talents in talent_combinations(klass):
+            Test(talents, group=grp, args=[('talents', talents)])
+
+
+def test_covenants(klass: str, path: str):
+    try:
+        from simc_support.game_data.WowSpec import get_wow_spec_from_combined_simc_name
+        from simc_support.game_data.Conduit import get_conduits_for_spec
+    except ImportError:
+        print("Error importing simc-support library for covenant test. Skipping test.")
         return
+
+    spec = get_wow_spec_from_combined_simc_name(klass)
+    conduits = get_conduits_for_spec(spec)
+    for fight_style in FIGHT_STYLES:
+        grp = TestGroup('{}/{}/covenants'.format(profile, fight_style),
+                        fight_style=fight_style, profile=path)
+        tests.append(grp)
+
+        # Test covenants
+        for covenant in COVENANTS:
```

Comment:
Considering this test requires simc_support library now anyway, I think we can drop our local `COVENANTS` array and use the one from the library.